### PR TITLE
Prepare for fly.io deploy - Add generated files and remove runtime references to Mix

### PR DESCRIPTION
**Pull request by joealba**
Mix is not available in production releases, so these references to Mix
prevented deploy to fly.io

### CODE REVIEW

File: lib/lunchbot_web/router.ex [hunk 1]
```diff
@@ -50,7 +50,7 @@ defmodule LunchbotWeb.Router do
   # If your application does not have an admins-only section yet,
   # you can use Plug.BasicAuth to set up some basic authentication
   # as long as you are also using SSL (which you should anyway).
-  if Mix.env() in [:dev, :test] do
+  if Application.get_env(:lunchbot, :env) in [:dev, :test] do
```
Comment (ericfreese) [inline lib/lunchbot_web/router.ex]: I think the phoenix generator spits out this code with `Mix.env()` referenced. We use releases on pineapple and don't have any problem with this usage of `Mix.env()`. I wonder if this is evaluated at compile time where mix is available? Basically... is this change really necessary?

File: lib/lunchbot/application.ex [hunk 2]
```diff
@@ -7,7 +7,7 @@ defmodule Lunchbot.Application do
 
   @impl true
   def start(_type, _args) do
-    unless Mix.env() == :prod do
+    unless Application.get_env(:lunchbot, :env) == :prod do
       Dotenv.load()
       Mix.Task.run("loadconfig")
     end
```
Comment (ericfreese) [inline lib/lunchbot/application.ex]: The [dotenv readme](https://github.com/avdi/dotenv_elixir/blob/d6fd3f327173fe18a455203987da95ef9f6cd4c5/README.md) says that it's not compatible with elixir releases. I assume you're using elixir releases since mix isn't available. Maybe it would be better to move away from dotenv and just load these env vars manually in `config/runtime.exs`? In fact I wonder if you could just move the `Dotenv.load()` and `Mix.Task.run("loadconfig")` calls to `config/runtime.exs` and not have to bother storing `config_env()` in the application config?

File: lib/lunchbot_web/router.ex (hunk 1 above)
Comment (joealba) [reply lib/lunchbot_web/router.ex]:
```
2022-07-14T13:53:07.000 [info]     ** (EXIT) an exception was raised:
2022-07-14T13:53:07.000 [info]         ** (UndefinedFunctionError) function Mix.env/0 is undefined (module Mix is not available)
```

https://community.fly.io/t/function-mix-env-0-is-undefined-module-mix-is-not-available/4181

Trying to deploy lunchbot to fly.io, and I'm running into the error described above.  And the answer above says that compiled releases do not include Mix.

Comment (ericfreese) [reply lib/lunchbot_web/router.ex]: Could that error be referring to the `Mix.env()` usage in `Application.start`? I don't see anything in the error message that indicates that this specific usage of `Mix.env()` is the problem

File: lib/lunchbot/application.ex (hunk 2 above)
Comment (joealba) [reply lib/lunchbot/application.ex]: I was assuming the dotenv was only being used in dev environments (local dev and test), so I was going to keep that in place for now.

File: lib/lunchbot_web/router.ex (hunk 1 above)
Comment (joealba) [reply lib/lunchbot_web/router.ex]: Yes, the error certainly seems to be referring to `application.ex`:  `2022-07-14T13:53:08.000 [info] Kernel pid terminated (application_controller) ({application_start_failure,lunchbot,{bad_return,{{'Elixir.Lunchbot.Application',start,[normal,[]]},{'EXIT',{undef,[{'Elixir.Mix',env,[],[]},{'Elixir.Lunchbot.Application',start,2,[{file,"lib/lunchbot/application.ex"},{line,10}]},{application_master,start_it_old,4,[{file,"application_master.erl"},{line,293}]}]}}}}})`

I was assuming that the similar calls in the router were going to lead to the same error as well.  

This is my first time tapping on an Elixir project.  :)

File: lib/lunchbot/application.ex (hunk 2 above)
Comment (ericfreese) [reply lib/lunchbot/application.ex]: You could still have an env check in `config/runtime.exs`

```ex
# In `config/runtime.exs`
unless config_env() == :prod do
  Dotenv.load()
  Mix.Task.run("loadconfig")
end
```

I see the readme for dotenv recommends placing them in Application.start but it also says it's not compatible with environments without Mix. Runtime config seems to me like it would be a better place to put it. And it's relatively new, so I wouldn't be surprised if the dotenv readme was just out of date. Actually, I just looked it up and there is an issue in the github repo along these lines: https://github.com/avdi/dotenv_elixir/issues/38

File: lib/lunchbot_web/router.ex (hunk 1 above)
Comment (ericfreese) [reply lib/lunchbot_web/router.ex]: > I was assuming that the similar calls in the router were going to lead to the same error as well.

Yah makes sense, just wanted to provide perspective that I don't think this is the case.

> This is my first time tapping on an Elixir project. :)

It's weird having this distinction between compile-time and run-time. I think in this case, that the `Mix.env()` calls in `Router` will be called at compile time, where mix _is_ available :)

Comment (bencates) [reply lib/lunchbot_web/router.ex]: This slightly verbose construction is also safe to use:

```elixir
if function_exported?(Mix, :env, 0) && Mix.env() == :dev do
  # dev stuff
end
```

File: lib/lunchbot/application.ex (hunk 2 above)
Comment (RVRX) [reply lib/lunchbot/application.ex]: > I assume you're using elixir releases since mix isn't available

Haven't used it before but I think running [`mix release`](https://hexdocs.pm/mix/1.13/Mix.Tasks.Release.html) in a dev environment creates a standalone app for production that might package mix?

**Comment by joealba** [comment]
👍  Thanks!  I've removed the changes to the router, and I've moved the dotenv load into config/runtime.

File: lib/lunchbot/application.ex (hunk 2 above)
Comment (bencates) [reply lib/lunchbot/application.ex]: It creates a standalone compiled app that does _not_ include `Mix` (or any other `:dev` dependencies).

**Review by chewnoill — APPROVED** [review]
🚀
